src/i_winmusic.c

```diff
@@ -230,73 +230,77 @@ static void StreamOut(void)
 {
     native_event_t native_event;
-    native_event.dwDeltaTime = time;
+    native_event.dwDeltaTime = delta_time;
     native_event.dwStreamID = 0;
     native_event.dwEvent = MAKE_EVT(status | channel, param1, param2, MEVT_SHORTMSG);
     WriteBuffer((byte *)&native_event, sizeof(native_event_t));
 }
 
-static void SendChannelMsg(int time, const midi_event_t *event, boolean use_param2)
+static void SendChannelMsg(unsigned int delta_time, const midi_event_t *event,
+                           boolean use_param2)
 {
-    SendShortMsg(time, event->event_type, event->data.channel.channel,
+    SendShortMsg(delta_time, event->event_type, event->data.channel.channel,
                  event->data.channel.param1,
                  use_param2 ? event->data.channel.param2 : 0);
 }
 
-static void SendLongMsg(int time, const byte *ptr, int length)
+static void SendLongMsg(unsigned int delta_time, const byte *ptr,
+                        unsigned int length)
 {
     native_event_t native_event;
-    native_event.dwDeltaTime = time;
+    native_event.dwDeltaTime = delta_time;
     native_event.dwStreamID = 0;
     native_event.dwEvent = MAKE_EVT(length, 0, 0, MEVT_LONGMSG);
     WriteBuffer((byte *)&native_event, sizeof(native_event_t));
     WriteBuffer(ptr, length);
     WriteBufferPad();
 }
 
-static void SendNOPMsg(int time)
+static void SendNOPMsg(unsigned int delta_time)
 {
     native_event_t native_event;
-    native_event.dwDeltaTime = time;
+    native_event.dwDeltaTime = delta_time;
     native_event.dwStreamID = 0;
     native_event.dwEvent = MAKE_EVT(0, 0, 0, MEVT_NOP);
     WriteBuffer((byte *)&native_event, sizeof(native_event_t));
 }
 
-static void SendDelayMsg(int time_ms)
+static void SendDelayMsg(unsigned int time_ms)
 {
     // Convert ms to ticks (see "Standard MIDI Files 1.0" page 14).
-    int time_ticks = (float)time_ms * 1000 * timediv / tempo + 0.5f;
-    SendNOPMsg(time_ticks);
+    const unsigned int ticks = (float)time_ms * 1000 * timediv / tempo + 0.5f;
+    SendNOPMsg(ticks);
 }
 
-static void UpdateTempo(int time, const midi_event_t *event)
+static void UpdateTempo(unsigned int delta_time, const midi_event_t *event)
 {
     native_event_t native_event;
 
     tempo = MAKE_EVT(event->data.meta.data[2], event->data.meta.data[1],
                      event->data.meta.data[0], 0);
 
-    native_event.dwDeltaTime = time;
+    native_event.dwDeltaTime = delta_time;
     native_event.dwStreamID = 0;
     native_event.dwEvent = MAKE_EVT(tempo, 0, 0, MEVT_TEMPO);
     WriteBuffer((byte *)&native_event, sizeof(native_event_t));
 }
 
-static void SendManualVolumeMsg(int time, int channel, int volume)
+static void SendManualVolumeMsg(unsigned int delta_time, byte channel,
+                                byte volume)
 {
-    int scaled_volume = volume * volume_factor + 0.5f;
-    SendShortMsg(time, MIDI_EVENT_CONTROLLER, channel,
+    const byte scaled_volume = (byte)(volume * volume_factor + 0.5f) & 0x7F;
```

Comment:
What happens if this gets truncated? Should it saturate to 0x7F instead?

Reply:
See https://github.com/chocolate-doom/chocolate-doom/pull/1598/commits/1b0c89ea5ec365b6230b707562754348190ae8ba